### PR TITLE
Optimize audit log processing by filtering files based on session timestamp

### PR DESCRIPTION
**Pull request by rlobillo**
##### Short description:
Optimize audit log processing and mov back `test_cnv_pod_security_violation_audit_logs` to IUO tier2 from IUO tier3.

##### More details:
Before this patch,audit log tests were timing out after 180s because they processed all historical audit log files (~33 files per node, potentially gigabytes of data). This caused test_cnv_pod_security_violation_audit_logs to be moved to tier3.

##### What this PR does / why we need it:
The solution is to filter audit log files by timestamp BEFORE processing them, keeping only files relevant to the current test session:

1. Parse timestamps from rotated log file names (audit-YYYY-MM-DDTHH-MM-SS.mmm.log)
2. Include only:
   - The immediately previous file before session start (buffer for edge cases)
   - All files with timestamps >= session start
   - The active audit.log file
3. Use UTC timezone to match audit log file naming convention

##### Implementation details:

- Add session_start_time fixture to capture test session start in UTC
- Modify audit_logs fixture to filter files by timestamp:
  - Parse timestamps from rotated log file names
  - Filter to keep only relevant files (previous + recent + active)
  - Add logging to show files processed vs total
  - Use timezone-aware comparison to avoid local/UTC mismatch

- Update pod_security_violations_apis_calls fixture to use filtered audit logs
- Fix boolean logic bug in CNAO exception filter (line 64-67): Before: ('container "cni-plugins"' or f'container "{BRIDGE_MARKER}"' in pod_audit_violations) After: ('container "cni-plugins"' in pod_audit_violations or f'container "{BRIDGE_MARKER}"' in pod_audit_violations)

##### Which issue(s) this PR fixes:
This benefits both test_cnv_pod_security_violation_audit_logs and test_deprecated_apis_in_audit_logs. Should allow removing quarantine from test_deprecated_apis_in_audit_logs (CNV-76514)

##### Special notes for reviewer:
N/A
##### jira-ticket:
https://issues.redhat.com/browse/CNV-71816

## Summary by CodeRabbit

* **Tests**
  * Improved audit-log filtering: session-aware timestamps, better handling of rotated vs active logs, and clearer processing messages for more reliable test results.
  * Updated pod security test wording and reclassified one test from tier3 to tier2.

* **Chores**
  * Added a session-level timestamp fixture for tests.
  * Enhanced type annotations and docstrings in utilities for clarity and maintainability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by openshift-virtualization-qe-bot-4** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

## Welcome! 🎉

This pull request will be automatically processed with the following features:

### 🔄 Automatic Actions
* **Reviewer Assignment**: Reviewers are automatically assigned based on the OWNERS file in the repository root
* **Size Labeling**: PR size labels (XS, S, M, L, XL, XXL) are automatically applied based on changes
* **Issue Creation**: A tracking issue is created for this PR and will be closed when the PR is merged or closed
* **Branch Labeling**: Branch-specific labels are applied to track the target branch
* **Auto-verification**: Auto-verified users have their PRs automatically marked as verified
* **Labels**: Enabled categories: `branch`, `can-be-merged`, `cherry-pick`, `has-conflicts`, `hold`, `needs-rebase`, `size`, `verified`, `wip`

### 📋 Available Commands

#### PR Status Management
* `/wip` - Mark PR as work in progress (adds WIP: prefix to title)
* `/wip cancel` - Remove work in progress status
* `/hold` - Block PR merging (approvers only)
* `/hold cancel` - Unblock PR merging
* `/verified` - Mark PR as verified
* `/verified cancel` - Remove verification status
* `/reprocess` - Trigger complete PR workflow reprocessing (useful if webhook failed or configuration changed)
* `/regenerate-welcome` - Regenerate this welcome message

#### Review & Approval
* `/lgtm` - Approve changes (looks good to me)
* `/approve` - Approve PR (approvers only)
* `/assign-reviewers` - Assign reviewers based on OWNERS file
* `/assign-reviewer @username` - Assign specific reviewer
* `/check-can-merge` - Check if PR meets merge requirements

#### Testing & Validation
 * `/retest tox` - Run Python test suite with tox
 * `/retest build-container` - Rebuild and test container image
 * `/retest all` - Run all available tests


#### Container Operations
* `/build-and-push-container` - Build and push container image (tagged with PR number)
  * Supports additional build arguments: `/build-and-push-container --build-arg KEY=value`

#### Cherry-pick Operations
* `/cherry-pick <branch>` - Schedule cherry-pick to target branch when PR is merged
  * Multiple branches: `/cherry-pick branch1 branch2 branch3`

#### Label Management
* `/<label-name>` - Add a label to the PR
* `/<label-name> cancel` - Remove a label from the PR

### ✅ Merge Requirements

This PR will be automatically approved when the following conditions are met:

1. **Approval**: `/approve` from at least one approver
2. **LGTM Count**: Minimum 2 `/lgtm` from reviewers
3. **Status Checks**: All required status checks must pass
4. **No Blockers**: No WIP, hold, conflict labels
5. **Verified**: PR must be marked as verified (if verification is enabled)

### 📊 Review Process

<details>
<summary><strong>Approvers and Reviewers</strong></summary>


**Approvers:**
 * dshchedr
 * myakove
 * rnetser
 * vsibirsk


**Reviewers:**
 * OhadRevah
 * RoniKishner
 * albarker-rh
 * dshchedr
 * geetikakay
 * hmeir
 * rlobillo
 * rnetser
 * vsibirsk


</details>

<details>
<summary><strong>Available Labels</strong></summary>

 * hold
 * verified
 * wip
 * lgtm
 * approve

</details>

### 💡 Tips

* **WIP Status**: Use `/wip` when your PR is not ready for review
* **Verification**: The verified label is automatically removed on each new commit
* **Cherry-picking**: Cherry-pick labels are processed when the PR is merged
* **Container Builds**: Container images are automatically tagged with the PR number
* **Permission Levels**: Some commands require approver permissions
* **Auto-verified Users**: Certain users have automatic verification and merge privileges

For more information, please refer to the project documentation or contact the maintainers.

**Comment by openshift-virtualization-qe-bot-3** [comment]
New container for quay.io/openshift-cnv/openshift-virtualization-tests:pr-3479 published

**Review by rnetser — DISMISSED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -12,7 +12,9 @@
 import shutil
 import subprocess
 import tempfile
```
[the remark this answers is not in the document]
Comment (rlobillo) [reply tests/conftest.py]: Validation has been run as detailed above. Please check "verified" comment.

[the remark this answers is not in the document]
Comment (rlobillo) [reply tests/conftest.py]: test_deprecated_apis_in_audit_logs is still quarantined and will be re-enabled as part of [CNV-76514](https://issues.redhat.com/browse/CNV-76514).
